lib/rules/no-object-constructor.js

```diff
@@ -80,6 +81,74 @@ module.exports = {
             return false;
         }
 
+        /**
+         * Determines whether a parenthesized object literal that replaces a specified node needs to be preceded by a semicolon.
+         * @param {ASTNode} node The node to be replaced.
+         * @returns {boolean} Whether a semicolon is required before the parenthesized object literal.
+         */
+        function needsSemicolon(node) {
+            const prevToken = sourceCode.getTokenBefore(node);
+
+            if (
+                !prevToken ||
+                prevToken.type === "Punctuator" && [":", ";", ">", "{", "=>", "++", "--"].includes(prevToken.value)
+            ) {
+                return false;
+            }
+
+            const prevNode = sourceCode.getNodeByRangeIndex(prevToken.range[0]);
+
+            if (isClosingParenToken(prevToken)) {
+                return ![
+                    "DoWhileStatement",
+                    "ForInStatement",
+                    "ForOfStatement",
+                    "ForStatement",
+                    "IfStatement",
+                    "WhileStatement",
+                    "WithStatement"
+                ].includes(prevNode.type);
+            }
+
+            if (isClosingBraceToken(prevToken)) {
+                return (
+                    prevNode.type === "BlockStatement" && prevNode.parent.type === "FunctionExpression" ||
+                    prevNode.type === "ClassBody" && prevNode.parent.type === "ClassExpression" ||
+                    prevNode.type === "ObjectExpression"
+                );
+            }
+
+            if (prevToken.type === "Identifier") {
+                return !["BreakStatement", "ContinueStatement"].includes(prevNode.parent.type);
+            }
+
+            if (prevToken.type === "Keyword") {
+
+                // Keywords that can immediately precede an ExpressionStatement node, mapped to the their node types.
+                const nodeTypesByKeyword = {
+                    __proto__: null,
+                    break: "BreakStatement",
+                    continue: "ContinueStatement",
+                    debugger: "DebuggerStatement",
+                    do: "DoWhileStatement",
+                    else: "IfStatement",
+                    return: "ReturnStatement",
+                    while: "WhileStatement",
```

Comment:
`while` seems unnecessary here as the `while` keyword of while statements cannot appear directly before an expression statement?